### PR TITLE
Spring Boot 2.0 anniversary meetup post

### PR DESCRIPTION
**Pull request by kencoenen**

### CODE REVIEW

File: _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+He talked about the new features while migrating a Spring Boot 1.x application to Spring Boot 2.0.
+
+# Spring Boot 2.0
+
+Stéphane gave us an overview of the new features in Spring Boot 2.0.
+It was kind of a summary of Phil Webb's announcement post <a href="https://spring.io/blog/2018/03/01/spring-boot-2-0-goes-ga" target="_blank">Spring Boot 2.0 goes GA</a>.
+
+Code says more than a thousand words.
+And like every talk I attended from Stéphane, he started live coding quite quickly.
+We migrated a a Spring Boot 1.x application to Spring Boot 2.0.
```
Comment (yannickdeturck) [inline _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md]: Delete an "a"

File: _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+The migration process is **very** simple.
+In short, these are the steps you have to follow:
+1. Change Spring Boot parent version number in your ``pom.xml``
+2. Replace changed property keys with the help of the ``spring-boot-properties-migrator`` module
```
Comment (yannickdeturck) [inline _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md]: changed -> deprecated?

File: _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+3. If you're working with passwords, define a ``PasswordEncoder``
+
+> When you use a lot of hookpoints and Spring Boot classes directly, eg. ``SpringBootServletInitializer``, migrating is a slightly more work.
```
Comment (yannickdeturck) [inline _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md]: migrating is slightly more work

File: _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+<blockquote class="twitter-tweet" data-cards="hidden" data-lang="nl"><p lang="en" dir="ltr">Spring Boot 2.0.1 available now<a href="https://t.co/WpYx0Pwff8">https://t.co/WpYx0Pwff8</a><a href="https://twitter.com/springboot?ref_src=twsrc%5Etfw">@springboot</a> <a href="https://twitter.com/springcentral?ref_src=twsrc%5Etfw">@springcentral</a> <a href="https://twitter.com/ProjectReactor?ref_src=twsrc%5Etfw">@ProjectReactor</a></p>&mdash; Stéphane Nicoll (@snicoll) <a href="https://twitter.com/snicoll/status/981899920750850049?ref_src=twsrc%5Etfw">5 april 2018</a></blockquote>
+<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+<br/>
+
+## Property keys and the properties migrator
+
+### Property key changes
+
+When you upgrade, you will get compilation errors saying there are unknown properties in your properties files.
+Some property keys are changed and won't work anymore.
```
Comment (yannickdeturck) [inline _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md]: Sounds better I think: 
Some property keys have been deprecated ...

File: _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+
+| Old property  | New property | 
+| ------------- | -------------|
+| ``spring.datasource.initialize`` | ``spring.datasource.initialization-mode`` |
+| ``endpoints.health.path`` | ``management.endpoints.web.path-mapping`` |
+
+All major IDE's, eg. IntelliJ, Netbeans and STS, will give you nice information about the newer property key.
```
Comment (yannickdeturck) [inline _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md]: IDEs

File: _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+The Spring Boot Properties Migrator looks for configuration properties it can replace and provides the programmer with feedback.
+For example, ``endpoints.health.path`` will automatically be replaced by the new ``management.endpoints.web.path-mapping`` property.
+
+Spring Boot 2.0 will also tell you you’re using an old key when you provide old keys as **environment variable**.
+It will not tell you the line on which you defined the property, but it will give an appropriate message.
+
+Some properties cannot be fixed automatically:
+
+* ``The security auto-configuration is no longer customizable``
+* Replacement key 'spring.datasource.initialization-mode' uses an incompatible target type — String was replaced by an enumeration.
```
Comment (yannickdeturck) [inline _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md]: Also wrap this one into backticks?

File: _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+``private Duration delay = Duration.ofSeconds(3)``
+
+The ``hello.delay`` autocompletion now also shows the unit behind the value, eg. ``hello.delay=3s``.
+
+### Spring Boot Dev Tools enhancements
+
+DevTools is a feature in Spring Boot which adds nice development features.
+
+Spring created its own small LiveReload server, with a reload function.
+The communication protocol of LiveReload is opensource.
```
Comment (yannickdeturck) [inline _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md]: open source

File: _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+When you start the app, it also starts the LiveReload server.
+Something in your application watches the classpath for changes.
+If you change a template or a configuration property, Spring Boot will pick up the change, restart the Spring context and notify LiveReload.
+The restart only takes about 1 to 3 seconds, because the JVM is still hot.
+
+Spring Boot Dev Tools is not new feature from Spring Boot 2, but the Spring developers have added some enhancements.
```
Comment (yannickdeturck) [inline _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md]: is not **a** new feature

File: _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+
+Stéphane tweets the day after that the Java 10 build was fixed.
+
+<blockquote class="twitter-tweet" data-lang="nl"><p lang="en" dir="ltr">Does <a href="https://twitter.com/springboot?ref_src=twsrc%5Etfw">@springboot</a> 2 support Java 10?<br><br>A picture is worth a thousand words <a href="https://t.co/5jpG27gJnE">pic.twitter.com/5jpG27gJnE</a></p>&mdash; Stéphane Nicoll (@snicoll) <a href="https://twitter.com/snicoll/status/982170678920003585?ref_src=twsrc%5Etfw">6 april 2018</a></blockquote>
+<script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+# The meetup
+
+The Spring Boot 2.0 Anniversary Meetup event started at 18h at the Ordina headquarters in Mechelen.
+Since you cannot learn on an empty stomach, we started off with belgian French fries.
```
Comment (yannickdeturck) [inline _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md]: Belgian

File: _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md (hunk 5 above)
Comment (fredericghijselinck) [inline _posts/2018-04-05-spring-boot-2-0-anniversary-meetup.md]: ... will inform you about the newer ...

**Pull request merged**